pkg/app/bootstrap.go

```diff
@@ -315,3 +331,29 @@ func buildkit(clicontext *cli.Context) error {
 
 	return nil
 }
+
+func updateConfigWithRegistry(clicontext *cli.Context, config *buildkitutil.BuildkitConfig) error {
+	// Parse registry strings
+	dirSet := make(map[string]bool)
+	for _, reg := range strings.Split(clicontext.String("registry"), ";") {
+		parts := strings.Split(reg, ",")
+		if len(parts) != 4 {
+			return fmt.Errorf("invalid registry string: %s", reg)
+		}
+
+		config.Registry = append(config.Registry, strings.Split(parts[0], "=")[1])
+		caPath := strings.Split(parts[1], "=")[1]
+		config.Ca = append(config.Ca, caPath)
+		config.Key = append(config.Key, strings.Split(parts[2], "=")[1])
+		config.Cert = append(config.Cert, strings.Split(parts[3], "=")[1])
```

Comment:
This has an assumption: the order has to be `register,ca,key,cert` but this has not been validated.

BTW, it seems these 3 files have to be in the same directory as the CA path. Otherwise the bind won't work.

Reply:
yeah so i think passing a json string might be better